Python/sysmodule.c

```diff
@@ -3358,6 +3358,14 @@ make_impl_info(PyObject *version_info)
     if (res < 0)
         goto error;
 
+    value = PyUnicode_FromString(Py_GetPlatform());
```

Comment:
Actually, by reading the issue, `_architecture` should reflect the `$(ArchName)` variable and not the `platform.implementation`. It appears that it's also only for Windows. So this fix should really take what `ArchName` contains and should only be present for Windows platforms.

`Py_GetPlatform()` only returns the same value as `sys.platform` so this is not the correct value to store.

Reply:
Yes, I noticed that what he seems to return is not the architecture but the system.